post_to_mastodon.py

```
@@ -86,15 +92,49 @@
+# first, we need to find the CHANGELOG.md file
+changelog_path = None
+for root, dirs, files in os.walk(os.getcwd()):
+    if "CHANGELOG.md" in files:
+        changelog_path = os.path.join(root, "CHANGELOG.md")
+
+release_notes = ""
+if changelog_path:
+    release_notes = ""
+    # now, try to extract the release notes
+    with open(changelog_path, "r") as changelog_file:
+        changlog_file.readline()  # skip the first line
```

Ruff (0.11.9)

Undefined name `changlog_file`

(F821)